tests/e2e/fixtures/storage_efs_dependencies.py

```diff
@@ -229,23 +247,119 @@ def on_create():
         efs_pv["metadata"]["name"] = claim_name
         write_cfg(efs_pv, efs_pv_filepath)
 
-        # Add the namespace to the pvc.yaml file
+        # Update the values in the pvc.yaml file
         efs_pvc = load_cfg(efs_pvc_filepath)
-        efs_pvc["metadata"]["namespace"] = DEFAULT_NAMESPACE
+        efs_pvc["metadata"]["namespace"] = DEFAULT_USER_NAMESPACE
         efs_pvc["metadata"]["name"] = claim_name
         write_cfg(efs_pvc, efs_pvc_filepath)
 
+        # Update the values in the permissions file
+        # Statically provisioned volume needs extra permissions
+        efs_permission = load_cfg(efs_permissions_filepath)
+        efs_permission["metadata"]["namespace"] = DEFAULT_USER_NAMESPACE
+        efs_permission["metadata"]["name"] = "permissions" + claim_name
+        efs_permission["spec"]["template"]["spec"]["volumes"][0][
+            "persistentVolumeClaim"
+        ]["claimName"] = claim_name
+        write_cfg(efs_permission, efs_permissions_filepath)
+
         kubectl_apply(efs_sc_filepath)
         kubectl_apply(efs_pv_filepath)
         kubectl_apply(efs_pvc_filepath)
+        kubectl_apply(efs_permissions_filepath)
 
         efs_claim["claim_name"] = claim_name
 
     def on_delete():
         kubectl_delete(efs_pvc_filepath)
         kubectl_delete(efs_pv_filepath)
         kubectl_delete(efs_sc_filepath)
+        kubectl_delete(efs_permissions_filepath)
 
     return configure_resource_fixture(
         metadata, request, efs_claim, "efs_claim", on_create, on_delete
     )
+
+
+@pytest.fixture(scope="class")
+def dynamic_provisioning(metadata, region, request, cluster):
+    claim_name = rand_name("efs-claim-auto-dyn-")
+    efs_pvc_filepath = (
+        "../../docs/deployment/add-ons/storage/efs/dynamic-provisioning/pvc.yaml"
+    )
+    efs_sc_filepath = (
+        "../../docs/deployment/add-ons/storage/efs/dynamic-provisioning/sc.yaml"
+    )
+    efs_permissions_filepath = (
+        "../../docs/deployment/add-ons/storage/notebook-sample/set-permission-job.yaml"
+    )
+    efs_auto_script_filepath = (
+        "../../docs/deployment/add-ons/storage/efs/auto-efs-setup.py"
+    )
+    efs_claim_dyn = {}
+    efs_client = get_efs_client(region)
+
+    def on_create():
+        # Run the automated script to create the EFS Filesystem and the SC
+        efs_auto_script_absolute_filepath = os.path.join(
+            os.path.abspath(sys.path[0]), "../" + efs_auto_script_filepath
+        )
+        st = os.stat(efs_auto_script_filepath)
+        os.chmod(efs_auto_script_filepath, st.st_mode | stat.S_IEXEC)
+        subprocess.call(
+            [
+                "python",
```

Comment:
when we plugin these scripts into tests, I would highly recommend natively integrating them by moving it under `e2e/utils/efs` directory instead of calling python script using subprocess. Following are the benefits:

- validate that `requirements.txt` for the script is upto date since tests and script use same file
- reduce a lot of code duplication related to creating clients, kustomize, kubectl, service accounts, printing banners which increases the readability. I think you can cut down the script by 30%+
  - see difference between [scriptv1](https://github.com/awslabs/kubeflow-manifests/blob/e4a2aa55e0e40c1228f45698ca3e7dbb6129934d/docs/deployment/rds-s3/secrets/secrets-setup.py) vs [scriptv2](https://github.com/awslabs/kubeflow-manifests/blob/4170afed2da01b5c13b01107b0688a7cd7583c00/tests/e2e/utils/notebooks/setup_secrets_access.py)
- standardisation of running in terms of different deployment docs
- potentially better error handling for tests
- some of the constants defined in this method can go into the respective file

Reply:
I did consider this. However, the idea here is to "test the script" the way the user will use it and not to automate the resource creation for tests. This automation was already in place using fixtures which is a more pythonic way of testing. If I call only specific methods from the script, I am not testing it in entirety. In that case the script need not be a part of this EFS integration at all and we should consider running separate tests for these user facing scripts.

Reply:
Fair point. I think its easy to find a balance here, one way is to create a high level driver function which both the test and script call once they have all arguments, but I am open to calling via subprocess as well

For most of the other issues, we move the script to utils. it'll be easy to remove code duplication, avoid the problem with multiple requirements file and also wouldnt need a `--directory` which can be confusing w.r.t what files and directory structure are needed here

Reply:
I moved the file to the tests directory and updated the docs README.

Reply:
IMHO this auto script is cleaner in the docs directory. I have moved it for now but let's discuss further tomorrow.

Reply:
moved to utils as discussed in standup.